### ISSUE TITLE
PHPCS incorrectly thinks a function argument with a default value is a global variable

### ISSUE THREAD

**Issue by Flimm** [WordPress-Coding-Standards/WordPress-Coding-Standards#1275]
Take this example:

```php
<?php

add_shortcode( 'hello', function( $attrs, $content = null ) {

} );
```

PHPCS will give this error:

```
 3 | ERROR | Variables defined by a theme/plugin should start with the theme/plugin prefix. Found: "$content". (WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedVariableFound)
```

This error is incorrect. `$content` is not a global, therefore, it does not need to be prefixed with the theme/plugin prefix. I think that the fact that this is an anonymous function is throwing it off.

I'm using PHPCS 3.2.2, Wordpress-Coding-Standards 0.14.0, and PHP version 7.1.12.

**Comment by jrfnl** [comment]
Thanks for reporting this. I'll look into it.

At the same time, your example code highlights a seconds issue with the sniff - it currently does not check whether shortcodes being added are prefixed (which they should be). I've added that to the feature request in #929

**Comment by Flimm** [comment]
@jrfnl A bug report acknowledged and fixed within 24 hours. Dat is geweldig! Thank you @jrfnl , @GaryJones and @JDGrimes .

**Comment by jrfnl** [comment]
You're welcome :grinning:
